File: src/lib/sRAD/gui/sComponent/SFrame.java
package lib.sRAD.gui.sComponent;

import lib.sRAD.gui.component.MainBar;
import lib.sRAD.gui.component.Theme;
import lib.sRAD.gui.component.VentanaEmergente;

import javax.swing.*;
import javax.swing.border.Border;
import java.awt.*;

import static lib.sRAD.gui.component.Resource.blackBorderTransparent;

public class SFrame extends JFrame {

    //ajustes pre-establecidos
    public static final int ESTANDAR = 1; //configuración que uso en mis proyectos

    /**
     * Constructor especializado
     * @param type
     */
    public SFrame(int type) {
        setProperties(type);
    }

    /**
     * Constructor de javax.swing.JFrame
     */
    public SFrame() {
        super();
    }

    /**
     * Create a new decorated frame
     * @param width width of frame
     * @param height height of frame
     */
    public SFrame(int width, int height) {
        setProperties(width, height, Theme.fBg, true, blackBorderTransparent, null, true);
    }

    public void setProperties(int type) {
        if(type == ESTANDAR) {
            setProperties();
        }
    }

    /**
     * Default properties of frame
     */
    public void setProperties() {
        setProperties(1280, 720, Theme.fBg, true, blackBorderTransparent, null, true);
    }

    public void setProperties(int width, int height) {
        setProperties(width, height, Theme.fBg, true, blackBorderTransparent, null);
    }

    public void setProperties(int width, int height, Color background) {
        setProperties(width, height, background, true, blackBorderTransparent, null);
    }

    public void setProperties(int width, int height, Color background, Boolean undecorated, Border border, Component relativeLocation) {
        setSize(width, height);
        getContentPane().setBackground(background);
        setUndecorated(undecorated);
        rootPane.setBorder(border);
        setLocationRelativeTo(relativeLocation);
        setLayout(null);
    }

    public void setProperties(int width, int height, Color background, Boolean undecorated, Border border, Component relativeLocation,
                              Boolean visible) {
        setProperties(width, height, background, undecorated, border, relativeLocation);
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        setVisible(visible);
    }

    public void setMainBar(String title) {
        setMainBar(title, "resources/sRAD/exampleLogo.png");
    }

    public void setMainBar(String title, String pathLogo) {
        MainBar mainBar = new MainBar(this);
        mainBar.setTitle(title);
        mainBar.setLogo(new ImageIcon(pathLogo));
        add(mainBar);
        repaint();
    }

}

File: gui/sComponent/SFrame.kt
package lib.sRAD.gui.sComponent

import lib.sRAD.gui.component.MainBar
import lib.sRAD.gui.component.Theme
import lib.sRAD.gui.component.blackBorderTransparent
import java.awt.Color
import java.awt.Component
import java.awt.LayoutManager
import javax.swing.ImageIcon
import javax.swing.JFrame
import javax.swing.border.Border

open class SFrame(width: Int=1280, height: Int=720) : JFrame() {

    init  {
        setProperties(width, height)
    }

    fun setProperties(width: Int, height: Int, background: Color=Theme.fBg, undecorated: Boolean = true,
                      border: Border = blackBorderTransparent, relativeLocation: Component? = null,
                      visible: Boolean = false, defaultCloseOperation: Int = EXIT_ON_CLOSE, layout: LayoutManager? = null) {
        setSize(width, height)
        contentPane.background = background
        isUndecorated = undecorated
        rootPane.border = border
        setLocationRelativeTo(relativeLocation)
        setDefaultCloseOperation(defaultCloseOperation)
        setLayout(layout)
        isVisible = visible
    }

    fun setMainBar(title: String) {
        setMainBar(title, "src/main/resources/sRAD/exampleLogo.png", width)
    }

    fun setMainBar(title: String, pathLogo: String, width: Int) {
        val mainBar = if(width>100) {
            MainBar(this, width)
        }
        else {
            MainBar(this)
        }
        mainBar.setTitle(title)
        mainBar.setLogo(ImageIcon(pathLogo))
        add(mainBar)
        repaint()
    }

}